tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -389,3 +392,152 @@ def finalize(self, state):
+  information, which includes intermediate calculations, are held in a
+  `RunningCentralMomentsState` as returned via `initialize` and `update`
+  method calls.
+
+  #### References
+  [1]: Philippe Pebay. Formulas for Robust, One-Pass Parallel Computation of
+        Covariances and Arbitrary-Order Statistical Moments. _Technical Report
+        SAND2008-6212_, 2008.
+        https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2008/086212.pdf
+  """
+
+  def __init__(self, shape, moment, dtype=tf.float32):
+    """Instantiates this object.
+
+    Args:
+      shape: Python `Tuple` or `TensorShape` representing the shape of
+        incoming samples.
+      moment: Integer that represents the central moment of interest.
+      dtype: Dtype of incoming samples and the resulting statistics.
+        By default, the dtype is `tf.float32`. Any integer dtypes will be
+        cast to corresponding floats (i.e. `tf.int32` will be cast to
+        `tf.float32`), as intermediate calculations should be performing
+        floating-point division.
+    """
+    self.shape = shape
+    self.moment = moment
+    if dtype is tf.int64:
+      dtype = tf.float64
+    elif dtype.is_integer:
+      dtype = tf.float32
+    self.dtype = dtype
+
+  def initialize(self):
+    """Initializes an empty `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` representing a stream of no
+        inputs.
+    """
+    mean_stream = RunningMean(
+        self.shape, self.dtype
+    )
+    return RunningCentralMomentsState(
+        num_samples=0,
+        mean=mean_stream.initialize().mean,
+        sum_exponentiated_residuals=tf.zeros(
+            (self.moment,) + self.shape, self.dtype),
+    )
+
+  def update(self, state, new_sample):
+    """Update the `RunningCentralMomentsState` with a new sample.
+
+    Args:
+      state: `RunningCentralMomentsState` that represents the current
+        state of running statistics.
+      new_sample: Incoming `Tensor` sample with shape and dtype compatible with
+        those used to form the `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` with updated calculations.
+    """
+    n_2 = 1
+    n_1 = state.num_samples
+    n = tf.cast(n_1 + n_2, dtype=self.dtype)
+    delta_mean = new_sample - state.mean
+    mean_stream = RunningMean(
+        self.shape, self.dtype
+    )
+    new_mean = mean_stream.update(RunningMeanState(
+        tf.cast(state.num_samples, self.dtype), state.mean,
+    ), new_sample).mean
+    old_res = state.sum_exponentiated_residuals
+    new_sum_exponentiated_residuals = [tf.zeros(self.shape, self.dtype)]
+
+    # the following two nested for loops calculate equation 2.9 in Pebay's
+    # 2008 paper from smallest moment to highest.
+    for p in range(2, self.moment + 1):
+      summation = tf.zeros(self.shape, self.dtype)
+      for k in range(1, p - 1):
+        adjusted_old_res = ((-delta_mean / n) ** k) * old_res[p-k-1]
```

Comment:
Add spaces around binary ops.

Reply:
Ah right. Sorry for continuously missing these, it's a personal preference of mine and my pylint doesn't yell at me when I don't do it